src/services/communications/Ethernet_packets.py

```diff
@@ -0,0 +1,52 @@
+import struct
+
+
+class Packets:
+    def __init__(self,*packets): 
+        #packets is a list where first goes an ID uint16_t and then string with types
+        self._dict_packets = {}
+        for packet in packets:
+            if len(packet) < 1:
+                raise ValueError ("Every packet need a ID at least")
+            if not isinstance(packet[0],int) or not (0 <= packet[0] <= 0xFFFF):
+                raise ValueError(f"The id must be an uint16_t, {packet[0]}")
+            id = packet[0]
+            self._dict_packets[id] = list()
+            for data_type in packet[1:]:
+                self._dict_packets[id].append(data_type)
+
+    def serialize_packet(self,packet_id,*values):
+        if packet_id not in self._dict_packets:
+            raise KeyError(f"Packet ID {packet_id} not found in the map")
+        
+        expected_sizes = self._dict_packets[packet_id]
+        
+        if len(values) != len(expected_sizes):
+            raise ValueError(f"Expected {len(expected_sizes)} but received {len(values)}")
+        
+        binary_packet = struct.pack('<H',packet_id)
+        for value, data_type in zip(values,expected_sizes):
+            match data_type:
+                case "uint8" | "enum": 
```

Comment:
the enum type is a bit tricky, it will be given to you as `enum(variantA,variantB,variantC)` and as many variants that are present. In the parsing step of the packet descriptions you should strip the variants and store them in some way as to later verify the provided variant is in range

Reply:
In the example above you have three variants:
* `variantA` with value 0
* `variantB` with value 1
* `variantC` with value 2

Reply:
okay and way is the utility for that, due to  an enum  in the normal cases starts from 0 and every element is one more

Reply:
Actually, what you should do with enums after thinking a bit about it is to get from the user a string that is the enum variant, for example the user would do `serialize_packet(1, "variantA")`. And then what you should do inside serialize is, check that the string is a variant of the enum, calculate the representation (0 in the example) and then serialize that as uint8.

This way the code is much more readable and the user doesn't need to be concerned with converting between the variant and the number by hand.